pkg/sql/create_table.go

```diff
@@ -73,47 +73,91 @@ type createTableRun struct {
 	fromHeuristicPlanner bool
 }
 
-// storageParamType indicates the required type of a storage parameter.
-type storageParamType int
+// storageParamObserver applies a storage parameter to an underlying item.
+type storageParamObserver interface {
+	apply(evalCtx *tree.EvalContext, key string, datum tree.Datum) error
+}
 
-// storageParamType values
-const (
-	storageParamBool storageParamType = iota
-	storageParamInt
-	storageParamFloat
-	storageParamUnimplemented
-)
+type tableStorageParamObserver struct {
+	tblDesc *sqlbase.MutableTableDescriptor
```

Comment:
why does this need a table

Reply:
in a future state, you may alter the table descriptor based on the storage params.
(e.g. you actually want to alter autovacuum_enabled to set some sort of bool on the descriptor to tell the opt not to collect stats)